view/frontend/templates/order/payments.phtml

```diff
@@ -46,8 +46,8 @@
                                     </a>
                                 </li>
                                 <li>
-                                    <a class="action show-payment-methods" data-id="<?= $recurringPayment->getId() ?>">
-                                        <?= $block->escapeHtml(__('Change card')) ?>
+                                    <a href="#" class="action">
```

Comment:
Class show-payment-methods are required. Without this class, the popup will not open. Also, data-id is prepared for future features which is in implementation